x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/AnalyticsUsage.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.analytics;
+
+import org.elasticsearch.common.xcontent.ContextParser;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Tracks usage of the Analytics aggregations.
+ */
+public class AnalyticsUsage {
+    private final AtomicLong boxplotUsage = new AtomicLong(0);
+    private final AtomicLong cumulativeCardUsage = new AtomicLong(0);
+    private final AtomicLong stringStatsUsage = new AtomicLong(0);
+    private final AtomicLong topMetricsUsage = new AtomicLong(0);
+
+    public <C, T> ContextParser<C, T> trackBoxplot(ContextParser<C, T> realParser) {
```

Comment:
Having a `trackFoo` and `getFooUsage` for each aggregation seems like a lot of boiler plate to me.  If I was writing this, I'd make an enum of things to track, have an `EnumMap` of `AtomicLong`s, and just one `track` and one `getUsage` method which accept that enum as an additional argument.  `EnumMap` uses an efficient array implementation, so we wouldn't be paying much in overhead for that.   This would mean that adding a new counter to this class would just. require adding an entry to the enum, rather than also copying and modifying the track and get methods.  Thoughts?

Reply:
I like it!